cmd/swagger-codegen/main.go

```diff
@@ -0,0 +1,438 @@
+			path.Operations.Post.Permitted = true
+		}
+		if doc.Paths[swaggerPath]["put"].OperationID != "" && getVerb != "put" {
+			path.Operations.Put.Permitted = true
+		}
+
+		// Add endpoint to paths
+		parent := findDeepestPath(paths, searchPath)
+		if parent == nil {
+			paths = append(paths, &path)
+		} else {
+			if parent.Endpoint.TemplateURL == path.Endpoint.TemplateURL {
+				// we have multiple entries with the same path (e.g. when applying a URL override)
+				// merge the two entries
+				// TODO Consider checking if there is a clash when merging operations
+				if path.Operations.Get.Permitted {
+					copyOperationFrom(path.Operations.Get, &parent.Operations.Get)
+				}
+				if path.Operations.Delete.Permitted {
+					copyOperationFrom(path.Operations.Delete, &parent.Operations.Delete)
+				}
+				if path.Operations.Patch.Permitted {
+					copyOperationFrom(path.Operations.Patch, &parent.Operations.Patch)
+				}
+				if path.Operations.Post.Permitted {
+					copyOperationFrom(path.Operations.Post, &parent.Operations.Post)
+				}
+				if path.Operations.Put.Permitted {
+					copyOperationFrom(path.Operations.Put, &parent.Operations.Put)
+				}
+				parent.Children = append(parent.Children, path.Children...)
+				parent.SubPaths = append(parent.SubPaths, path.SubPaths...)
+			} else {
+				if countNameSegments(parent.Endpoint) == countNameSegments(path.Endpoint) {
+					// this is a child
+					parent.Children = append(parent.Children, &path)
+				} else {
+					// this is a sub-resource
+					parent.SubPaths = append(parent.SubPaths, &path)
+				}
+			}
+		}
+	}
+	return paths
+}
+
+func copyOperationFrom(from PathOperation, to *PathOperation) {
+	to.Permitted = from.Permitted
+	to.Endpoint = from.Endpoint
+	to.Verb = from.Verb
+}
+func countNameSegments(endpoint *endpoints.EndpointInfo) int {
+	count := 0
+	for _, segment := range endpoint.URLSegments {
+		if segment.Name != "" {
+			count++
+		}
+	}
+	return count
+}
+
+func writeHeader(w io.Writer) {
+	w.Write([]byte("package handlers\n"))
+	w.Write([]byte("\n"))
+	w.Write([]byte("func (e *SwaggerResourceExpander) getResourceTypes() []ResourceType {\n"))
+	w.Write([]byte("\treturn []ResourceType{\n"))
+}
+func writeFooter(w io.Writer) {
+	w.Write([]byte("\t}\n"))
+	w.Write([]byte("}\n"))
+}
+func writePaths(w io.Writer, paths []*Path, config *Config, prefix string) { // TODO want to not need config here
+	for _, path := range paths {
+		pathVerb := config.Overrides[path.Endpoint.TemplateURL].GetVerb
+		if pathVerb == "" {
+			pathVerb = "get"
+		}
+		if path.Operations.Get.Permitted {
+			fmt.Fprintf(w, "\t%s\tResourceType{\n", prefix)
+			getEndpoint := path.Operations.Get.Endpoint
+			if getEndpoint == nil {
+				getEndpoint = path.Endpoint
+			}
+			fmt.Fprintf(w, "\t%s\t\tDisplay:  \"%s\",\n", prefix, path.Name)
+
+			fmt.Fprintf(w, "\t%s\t\tEndpoint: mustGetEndpointInfoFromURL(\"%s\", \"%s\"),\n", prefix, getEndpoint.TemplateURL, getEndpoint.APIVersion)
```

Comment:
[Nit] Would this be more readable using go templates? Traefik uses them when outputting stuff, helm is based on them and I know the AKS Engine team used to use them. I guess the pro is that the output is more visual when editing. [This is generating a TOML doc using them](https://github.com/containous/traefik-extra-service-fabric/blob/master/servicefabric_tmpl.go) https://golang.org/pkg/text/template/

Reply:
Would you want this done before merging, or as a separate PR?

Reply:
Yeah sounds good I've tracked it with #64